### ISSUE TITLE
[Feature Request] new vue/no-unused-components default

### ISSUE THREAD

**Issue by johnleider** [vuejs/eslint-plugin-vue#1048]
This is a Feature Request to update the default value of https://eslint.vuejs.org/rules/no-unused-components.html#vue-no-unused-components . In practice, this is extremely annoying to use when debugging:

```vue
<template>
  <div>
    <!-- <my-component></my-component> -->

    <div>...</div>
  </div>
</template>

<script>
  export default {
    components: {
      MyComponent: () => import('./MyComponent'),
    },
  }
</script>
```

In this example, I'm wanting to test something by commenting out the component. This rules greets me with a black screen preventing me from seeing the app until I additionally comment out the component import.

I'm proposing a new default value of `warn`. It still promotes the developer to make good decisions, but doesn't get in the way of working.

![image](https://user-images.githubusercontent.com/9064066/73975054-2a27fe80-48eb-11ea-95cb-d214fb620d8d.png)

**Comment by danbaechtold** [comment]
Yes! Please!

**Comment by ota-meshi** [comment]
Thank you for this issue.

We do not configure rulesets for debugging purposes.

I recommend that you create your own rulesets for debugging.
I think it will include rules such as [no-unused-vars](https://eslint.org/docs/rules/no-unused-vars) rule.

**Comment by johnleider** [comment]
Fair enough but it’s still a terrible developer experience.
